tests/extendrtests/tests/testthat/test-test-submodules.R

```diff
@@ -21,3 +21,25 @@ test_that("Classes defined in submodules also works", {
     isTRUE(identical(x$me_owned(), x))
   )
 })
+
+test_that("Return the right externalptr", {
```

Comment:
Again, I would say I see here three distinct tests, perhaps it is worth splitting it into smaller pieces? (And also having all externalptr-related tests in a single file, not just some `test-submodules`)?

Reply:
I've split off the `externalptr` tests into separate rust modules, testthat file, and everything. Hope this satisifies this point.